### ISSUE TITLE
Making the documentation more friendly to beginners

### ISSUE THREAD

**Issue by faassen** [mobxjs/mobx-react-docz#99]
I'm a long term mobx and mobx-react user, but I find the current mobx-react documentation rather technical and difficult to comprehend. You're hit with a bunch of options and advanced concepts all at once, and the notion that you manage a lot of state in mobx *models* as opposed to the use of useLocalStore and friends is not really discussed early on.

My proposal is that I update the documentation with a focus on basic introduction and simple usage patterns before it dives into advanced options. Would that be accepted?

I want to make modern React with function components and hooks prominent (even though I'm less familiar with it), but also carve out a basic intro on class-based mobx-react (with which I'm very familiar).

**Comment by FredyC** [comment]
@faassen Thank you for the feedback, I totally agree with you. It was mostly just my personal attempt to collect thoughts on how I know MobX with React so it's rather opinionated and probably even lacking.

So if you want to dive in and improve, I am surely open to it. That's what open source is about, after all.

What I also wanted to achieve somehow was to provide a guide on both hooks and classes in some manner where all examples will be done in both ways for comparison and you can toggle some switch to see either. Alternatively, we could collect all class-related stuff that's different from a hook on one or two separate pages.

Just please instead of doing one big bulk PR try to split it into smaller parts so it's easier to review it.

I was even thinking of switching to Docusaurus instead of Docz, but if you don't have particular experience with either, don't worry about it. Markdown files can be always reused anywhere.

**Comment by faassen** [comment]
Thanks for the fast feedback and I really appreciate your openness! I'm going to spend some time over the next couple of weeks to get some things written.

Concerning a toggle to switch either: I think that's a neat idea but it's also quite challenging. I think I'm going to focus on writing two separate documents for now and see how similar they are first.

I'm not familiar either with Docz or Docusaurus.

I will keep in mind to keep PRs coming incrementally!

**Comment by faassen** [comment]
I've extended the document extensively and have now created a draft pull request

**Comment by faassen** [comment]
I'm giving up on this topic. I hope it can be resolved once MobX 6 is released by someone else.

**Comment by FredyC** [comment]
Sure, no worries. It's really bad timing now when we are nearing such a big change in the MobX world. All existing documentation will need heavy updates so it's really better to minimize it here when we know what's going to happen.

**Comment by faassen** [comment]
I disagree. I think this would have improved matters and I think it's really a shame you blocked it, but I don't want to waste my energy on this anymore.

**Comment by FredyC** [comment]
Blocked it :D I've reasoned with arguments and you just gave up. If you can come with a good reason why duplicating numerous decorators example is a good idea when it's becoming deprecated, I am open to it.
